lib/DocumentManager.php

```diff
@@ -175,4 +265,34 @@ private function getOptionsResolver($eventName)
 
         return $resolver;
     }
+
+    /**
+     * If the exception is an instanceof `DocumentManagerException` set the
+     * document manager name that threw.
+     *
+     * @param \Exception $e
+     * @param string $context
+     */
+    private function processException(\Exception $exception, $context)
```

Comment:
Why? It is a private method, so it's only called by this class. And changing it the way I've suggested would have the same result, without catching `\Exception`, which only rarely makes sense, and having "flatter" code in this method.